### ISSUE TITLE
meta-issue on OCaml forge deprecation

### ISSUE THREAD

**Issue by gasche** [ocaml-batteries-team/batteries-included#700]
The OCaml forge is getting deprecated in 2017.

We should stop relying on it, migrate what needs migrated, and try to find a place to archive content of historical interest that we don't expect to use or modify in the future.

- HTML webpages: tracked in #660 
- mailing-list archives: tracked in #699
- bug tracker ?
- release history? We have git tags, but keeping announces may be nice.

**Comment by aantron** [comment]
@gasche, me and @pmetzger would like to delete the ocamlforge page of Batteries. It still comes up first in Google results, which misleads people that might want to use Batteries by showing them out-of-date information, which then gives a very bad impression of the project and harms Batteries.

We looked through what's left on ocamlforge. My personal opinion is that everything that is there, is already being handled by GitHub better, in a way that is more familiar to a "typical" Batteries user (because GitHub is familiar).

I think we need to just:

- Copy/paste all the announcements texts out to a text file (http://forge.ocamlcore.org/news/?group_id=17).
- Save all the batteries-devel text archives linked on this page: https://lists.forge.ocamlcore.org/pipermail/batteries-devel/
- I couldn't find the old bug tracker, but if it's there, dump it somehow.

The Batteries home page (http://batteries.forge.ocamlcore.org/) doesn't seem to provide any information that couldn't be put neater into the README. It has outdated news, comments counts at 0, outdated links, etc. I'd say, since it's not well-maintained, to simply delete it and commit to using the GitHub README only. If that becomes a problem in the future, it's always possible to set up some new web pages according to whatever process you think will be maintainable, and will solve the problems you are having with the README.

We can take all the things dumped above, and attach them as binary artifacts to a GitHub tag, so that they will be hosted online, and not hidden on someone's computer. I don't think it's necessary to actually import the mailing list or old bug tracker. I suspect nobody ever accesses those things. In the unlikely event a maintainer needs to do an archeological dig through them in the future, though, they will still be there, even if in somewhat inconvenient form. I think this is the right point in the effort/benefit space.

After that, we should delete the ocamlforge Batteries website, if possible.

**Comment by aantron** [comment]
If there are old releases of Batteries still hosted on forge, we can download those and also attach them as binaries to a GitHub tag, then point opam to them.

**Comment by pmetzger** [comment]
Maybe the API docs etc. could get moved to a github.io page?

**Comment by aantron** [comment]
(We found them, http://ocaml-batteries-team.github.io/batteries-included/hdoc2/)

**Comment by gasche** [comment]
Thanks for looking at this!

I agree that moving off the forge is the reasonable thing to do now.

- I can take the remaining main page content and move it to a README.md.

- The announcement texts could be just committed in a `doc/announces/` directory in the git repo. I can take care of it.

- I don't remember if we used to have a bugtracker on the forge, but I don't think so.

- On the other hand, I would like the preserve the archives of the batteries-devel and the batteries-discuss mailing-lists. I would be happy if they were recorded on public-inbox, or in some other reasonable way (I would be happier if people could access the archives from the web, rather than having to fetch and search a text dump). Unfortunately I have no expertise on how to do this. Would someone be willing to help? Maybe we could discuss this on the dedicated issue, #699.

**Comment by pmetzger** [comment]
> The announcement texts could be just committed in a doc/announces/ directory in the git repo. I can take care of it.

Future announcements can also be made on the discuss.ocaml.org site.

>  I would like the preserve the archives of the batteries-devel and the batteries-discuss mailing-lists. 

Maybe a github.io site?

**Comment by pmetzger** [comment]
No, I was more suggesting that the archives be downloaded, committed to a github repo, and maybe made available via a github.io site. But perhaps this is more than is needed and just having the files archived would be enough?
